### PR TITLE
tests: migrate test-infra to testify

### LINKED ISSUE #27874: migrate test-infra to testify for `tests` folder
* global_kill_test.go
* graceshutdown_test.go
* readonly_test.go

### PR DESCRIPTION
**Pull request by zwj-coder**
### What problem does this PR solve?

Issue Number: close #27874 

Problem Summary:  migrate test-infra to testify for `tests` folder

### What is changed and how it works?

What's Changed: code in `tests` folder

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks a lot for your admirable work @zwj-coder ! The next time you can try to request review by `/cc @reviewer` so that we can react on the pr more quickly.

File: tests/globalkilltest/global_kill_test.go [hunk 1]
```diff
@@ -396,125 +389,130 @@ func sleepRoutine(ctx context.Context, sleepTime int, conn *sql.Conn, connID uin
...
-	const sleepTime = 2
+	sleepTime := 2
 
 	// Test mysql client CTRL-C
 	// mysql client "CTRL-C" truncate connection id to 32bits, and is ignored by TiDB.
-	elapsed := s.killByCtrlC(c, port, sleepTime)
-	c.Assert(elapsed, GreaterEqual, sleepTime*time.Second)
+	elapsed := s.killByCtrlC(t, port, sleepTime)
+	log.Info("elapsed", zap.Any("elapsed", elapsed))
+	require.GreaterOrEqual(t, elapsed.Seconds(), float64(sleepTime))
```
Comment (xhebox) [inline tests/globalkilltest/global_kill_test.go:452]: There is `GreaterOrEqualf(t, ..., ...,  msg string, args ...interface{})`. You can use replace log with that. Logging will print something despite the result.

File: tests/globalkilltest/global_kill_test.go [hunk 2]
```diff
@@ -396,125 +389,130 @@ func sleepRoutine(ctx context.Context, sleepTime int, conn *sql.Conn, connID uin
...
 
 	// Test KILL statement
-	elapsed = s.killByKillStatement(c, db, db, sleepTime)
-	c.Assert(elapsed, Less, sleepTime*time.Second)
+	elapsed = s.killByKillStatement(t, db, db, sleepTime)
+	require.Less(t, elapsed.Seconds(), float64(sleepTime))
```
Comment (xhebox) [inline tests/globalkilltest/global_kill_test.go:456]: Why not `require.Less(t, elapsed, sleepTime*time.Second)`? Any reason for the change?

File: tests/readonlytest/readonly_test.go [hunk 3]
```diff
@@ -130,41 +134,43 @@ func (s *TestReadOnlySuit) TestRestrictionWithConnectionPool(c *C) {
...
 	}(conn)
 	time.Sleep(1 * time.Second)
 	timer := time.NewTimer(3 * time.Second)
 	_, err = s.db.Exec("set global tidb_restricted_read_only=1")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	// SUPER user can't write
 	_, err = s.db.Exec("insert into t values (1)")
-	c.Assert(err.Error(), Equals, ReadOnlyErrMsg)
+	require.NoError(t, err)
```
Comment (xhebox) [inline tests/readonlytest/readonly_test.go:174]: Please check if it is equal to `ReadOnlyErrMsg`. There is `Error(t TestingT, err error, msgAndArgs ...interface{})`. Do not change the logic of tests easily.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
@xhebox it seems we don't run these tests in PR tasks. I'll push a follow up for clean up some code for this one.

**Pull request merged**